### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by eamador1**
Since the last code review, the following changes were made on this project:
---

- Create a method called current_user in your ApplicationController to make current user data available to all controllers.
                 -It will return the first user from the database ✅

- Create forms to perform the following functions:
                 -Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController)  ✅
                 -Create a comment on behalf of the current_user (use the method that you created in your ApplicationController)  ✅

- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+class CommentsController < ApplicationController
+  before_action :set_user, only: [:create]
+  before_action :set_post
+
```
Comment (Lucas-Masaba) [inline app/controllers/comments_controller.rb:9]: - [ ] You are required to add a `new` action to this controller for example
```ruby
  def new
    @comment = Comment.new
  end
```

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @eamador1  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- Met most requirements for this milestone :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by eamador1** [comment]
Since the last code review, the following was done:

- Add new action to comments_controller

File: app/views/comments/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+user_post_comments_path(@user, @post)) do |form| %>
+ 
+  <div>
+    <%= form.label     :text %>
+    <%= form.text_area :text %>
+  </div>  
+  <div>
+    <%= form.submit "Add Comment" %>
+  </div>
+<% end %>
```
Comment (Whoistolu) [inline app/views/comments/_form.html.erb]: - [ ] [REQUIRED] Hello @eamador1 , it is good that you have added the new action to the comments controller but you need to create this form in the views/comments/new.html.erb file which accepts input for creating comments just like you did for the Post.

When I wanted to create a comment on the first post, I visited the correct route and got this error.

The error message "CommentsController#new is missing a template for request formats: text/html" indicates that there is no corresponding HTML view template for the new action in your CommentsController.

Screenshot 2024-01-12 at 03 58 46

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @eamador1 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/comments/_form.html.erb (hunk 2 above)
Comment (eamador1) [reply app/views/comments/_form.html.erb]: @Whoistolu I was not getting that error, but it does make sense. Thank you

File: app/views/comments/new.html.erb [hunk 3]
```diff
@@ -0,0 +1 @@
+<%= render 'form', comment: @comment %>
```
Comment (rloterh) [inline app/views/comments/new.html.erb]: - [ ] Good job on adding comment view form, however, although comments can be added via the post#show, an error still surfaces in an attempt to create a new form via the  `/users/1/posts/1/comments/new` route, like indicated in the previous review. Let's ensure to improve upon this aspect.

**Error in displaying comment view via users/1/posts/1/comments/new:**
image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Erwin, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁  🌴 🚦 🚗 🏫  🏡   _🏃 

## Highlights 👍  
- Post forms working as intended!
- Good ReadMe!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by eamador1** [comment]
Since the last code review, the following was done:
---

- add comment accessing form via the /users/1/posts/1/comments/new route

**Review by Ibizugbe — APPROVED** [review]
Hi @eamador1 👋🏼👋🏼👋🏼, 

approved


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights✅✅✅

- [x] Linters are passing👍🏼.
- [x] Forms to add comments have been created and can be added via the /users/1/posts/1/comments/new route👍🏼.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**